### PR TITLE
NDcPP update w/ TLSv1.3 proposal

### PR DESCRIPTION
**Pull request by kr15tyk**
FMT_SMF.1.1 - edited
FIA_X509_EXT.1.1/ITT - edited
FCS_TLSS_EXT.2.1 - edited
FCS_TLSS_EXT.2.4 - added
A.7.1.2 edited section #
B.3.1.1 added wording
B.3.1.6 added and edited wording

C. Extended Component Definitions
FCS_TLSS_EXT.2.1 - edited, no app notes included in C. ECD
FCS_TLSS_EXT.2.4 - added, no app notes included in C. ECD

### CODE REVIEW

**Review by ianhallbc — APPROVED** [review]

File: NDcPP_v2_2e.adoc [hunk 1]
```diff
@@ -1932,12 +1933,17 @@ _The use of X.509v3 certificates for TLS is addressed in FIA_X509_EXT.2.1. This
 
 *FCS_TLSS_EXT.2 TLS Server Support for Mutual Authentication*
 
-*FCS_TLSS_EXT.2.1* The TSF shall support TLS communication with mutual authentication of TLS clients using X.509v3 certificates.
+*FCS_TLSS_EXT.2.1* The TSF shall support TLS communication with mutual authentication of TLS clients using X.509v3 certificates and shall [selection:
+
+* _reject the connection if the client does either not provide a client certificate at all or the client certificate cannot be successfully validated by the TOE (except for override mechanisms that might be defined in FCS_TLSS_EXT.2.2) ('hard fail')_
```
Comment (plughy2) [inline NDcPP_v2_2e.adoc:1938]: Please change to:  "...reject the connection if the client either does not provide a client certificate at all, or the client certificate..."

File: NDcPP_v2_2e.adoc [hunk 2]
```diff
@@ -1965,6 +1971,34 @@ _If the identifier is not a FQDN, then the TSS shall describe how the identifier
...
+** _rsa_pss_pss with sha512(0x080b)_
+** _] and no other algorithms;_ +
+_]._
+
+*_Application Note {counter:appnote_count}_*
+
+_The selected ClientCertificateTypes and algorithms must be consistent with FCS_COP.1/SigGen and FCS_COP.1/Hash._ +
+_The TLS 1.2 Certificate Request message includes the algorithms as a list of  SignatureAndHashAlgorithms in the supported signature algorithms list. The signature algorithm is also specified in the certificate_types list._ +
+_The TLS 1.3 Certificate Request message includes the algorithms as a list of SignatureSchemes in the signature_algorithms extension._
+
```
Comment (plughy2) [inline NDcPP_v2_2e.adoc:2001]: Please check there is a line break at the end of each sentence of the Application Note.

File: NDcPP_v2_2e.adoc [hunk 3]
```diff
@@ -3811,7 +3849,10 @@ FIA_X509_EXT.1 X.509 Certificate Validation
 FIA_X509_EXT.2 X.509 Certificate Authentication
 ____
 
-*FCS_TLSS_EXT.2.1* The TSF shall support TLS communication with mutual authentication of TLS clients using X.509v3 certificates.
+*FCS_TLSS_EXT.2.1* The TSF shall support TLS communication with mutual authentication of TLS clients using X.509v3 certificates and shall [selection:
+
+* _reject the connection if the client does either not provide a client certificate at all or the client certificate cannot be successfully validated by the TOE (except for override mechanisms that might be defined in FCS_TLSS_EXT.2.2) ('hard fail')_
+* _accept the connection even if the client does not provide a client certificate at all, as long as a fall-back authentication is performed using one of the other authentication mechanisms defined in this cPP before any other TSF-mediated action is performed via this channel ('soft fail')_
```
Comment (plughy2) [inline NDcPP_v2_2e.adoc:3855]: Please change to: "...reject the connection if the client either does not provide a client certificate at all, or the client certificate..."

File: NDcPP_v2_2e.adoc (hunk 1 above)
Comment (kr15tyk) [reply NDcPP_v2_2e.adoc:1938]: @plughy2 Good catch on the wording. I've made that change here and in the word doc.

File: NDcPP_v2_2e.adoc (hunk 3 above)
Comment (kr15tyk) [reply NDcPP_v2_2e.adoc:3855]: @plughy2 I've made that change here and in the word doc.

File: NDcPP_v2_2e.adoc (hunk 2 above)
Comment (kr15tyk) [reply NDcPP_v2_2e.adoc:2001]: @plughy2 When I view this from my temporary branch, tls-110521-kk. I don't see an issue with formatting. Maybe we can discuss this on the call today.

Comment (kr15tyk) [reply NDcPP_v2_2e.adoc:2003]: @plughy2 I made edits that should have fixed this.

**Review by plughy2 — APPROVED** [review]

**Pull request merged**